src/main/java/Parser.java

```diff
@@ -0,0 +1,80 @@
+import task.TaskList;
+
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+public class Parser {
+    private final Scanner scanner;
+    private final TaskList taskList;
+
+    public Parser(TaskList taskList, Scanner scanner) {
+        this.scanner = scanner;
+        this.taskList = taskList;
+    }
+
+    public void handleInput(String input) {
+        while(!input.equalsIgnoreCase("bye")) {
+            // display task list
+            input = input.toLowerCase();
+            if(input.equals("list")) {
+                taskList.printTasks();
+                input = scanner.nextLine();
+                continue;
+            }
+
+            // mark tasks as done
```

Comment:
comments to explain each command follow coding style well and helpful in understanding the code